lib/identity_cache.rb

```diff
@@ -163,3 +163,5 @@ def fetch_in_batches(keys)
     end
   end
 end
+
+require 'identity_cache/railtie' if defined?(Rails)
```

Comment:
Rafael was mentioning that it's better to check for Rails::Railtie as Rails itself is sometimes manually defined inside non Rails application

Reply:
Interesting,[ the doc state to check for `defined?(Rails)`](https://api.rubyonrails.org/classes/Rails/Railtie.html), but I guess we could update it.

Do you mind if I merge as is, and eventually update after with the doc when Rafael get back? I need that change for a cleanup.

Reply:
Yeah sure , let's merge

Reply:
Yeah, let's update the doc.

If a gem is called rails-foo it will define `Rails` even if the rails framework is not required.

Reply:
https://github.com/rails/rails/pull/37127

Reply:
Interesting, I never hit any issue with `defined?(Rails)` for loading a Railtie before... any chance of `Rails::Railtie` not being required/defined by the time the gem loads, while `Rails` would be? I think not, but just wondering if that couldn't cause loading edge cases.

Reply:
`require "rails"` [loads `rails/application`](https://github.com/rails/rails/blob/6d68bb5f695414b1801c1c3952ef955a5b0b6c6a/railties/lib/rails.rb#L13), which [loads `rails/engine`](https://github.com/rails/rails/blob/6d68bb5f695414b1801c1c3952ef955a5b0b6c6a/railties/lib/rails/application.rb#L11), which [loads `rails/railtie`](https://github.com/rails/rails/blob/6d68bb5f695414b1801c1c3952ef955a5b0b6c6a/railties/lib/rails/engine.rb#L3). If `Rails` is defined by the `rails` gem, `Rails::Railtie` will be defined too.

Reply:
What @rafaelfranca suggests here are gems named `rails-somestuf`, with the following namespace:

```ruby
module Rails
  module SomeStuff
  end
end
```

They might or might not `require "rails"`. 

Honestly I've never been bit by this, but I get what Rafael means.

Reply:
nvm, I misunderstood the question you were answering to.

Reply:
Yeah. The rails-html-sanitizer gem already caused this project in a Sinatra app once. I remember the person even opened an issue on Rails about it.

Reply:
Right yeah, I can totally see how it could happen, just haven't been bitten by it before. Sounds good then, if Rails::Railtie is always there when rails is required. :+1: